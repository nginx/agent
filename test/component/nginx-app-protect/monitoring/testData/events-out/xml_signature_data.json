{
    "SecurityViolationEvent": {
        "PolicyName": "app_protect_default_policy",
        "SupportID": "4355056874564592516",
        "Outcome": "REJECTED",
        "OutcomeReason": "SECURITY_WAF_VIOLATION",
        "Method": "GET",
        "Protocol": "HTTP",
        "URI": "/",
        "Request": "GET /?a=\u003cscript\u003e HTTP/1.1\\r\\nHost: 127.0.0.1\\r\\nUser-Agent: curl/7.64.1\\r\\nAccept: */*\\r\\n\\r\\n",
        "RequestStatus": "blocked",
        "ResponseCode": "Blocked",
        "VSName": "1-localhost:1-/",
        "RemoteAddr": "127.0.0.1",
        "RemotePort": "61478",
        "ServerPort": "80",
        "Violations": "HTTP protocol compliance failed,Illegal meta character in value,Attack signature detected,Violation Rating Threat detected,Bot Client Detected",
        "ClientClass": "Untrusted Bot",
        "Severity": "critical",
        "BotSignatureName": "curl",
        "ViolationsData": [
            {
                "Name": "VIOL_ATTACK_SIGNATURE",
                "Signatures": [
                    {
                        "ID": "200021094",
                        "BlockingMask": "4",
                        "Buffer": "Connection: keep-alive\r\nHost: a.com\r\nUser-Agent: Java/1.7.0_51\r\nAccept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2\r\n\r\n",
                        "Offset": "37",
                        "Length": "16"
                    },
                    {
                        "ID": "200011034",
                        "BlockingMask": "2",
                        "Buffer": "pt: */*\r\nFoo: Authorization: %n%n%n%n\r\nFoo: echo\u003c!-- #echo\r\nX18",
                        "Offset": "29",
                        "Length": "6"
                    },
                    {
                        "ID": "200000179",
                        "BlockingMask": "3",
                        "Buffer": "7\r\nAccept: */*\r\nFoo: Authorization: %n%n%n%n\r\nFoo: echo\u003c!-- #ec",
                        "Offset": "21",
                        "Length": "23"
                    },
                    {
                        "ID": "200004106",
                        "BlockingMask": "2",
                        "Buffer": "zation: %n%n%n%n\r\nFoo: echo\u003c!-- #echo\r\nX1892: echo\r\nX1893: '\u003c!-",
                        "Offset": "27",
                        "Length": "10"
                    }
                ]
            }
        ]
    }
}