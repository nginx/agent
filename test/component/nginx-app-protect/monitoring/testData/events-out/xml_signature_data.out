Žapp_protect_default_policy4355056874564592518"REJECTED*SECURITY_WAF_VIOLATION:GETBHTTPR/ZcGET /?a=<script> HTTP/1.1\r\nHost: 10.146.179.119\r\nUser-Agent: curl/7.64.1\r\nAccept: */*\r\n\r\njblockedr0Štestenv-ad595bed-data-1š1-localhost:1-/¢172.18.9.78ª61478²80ºŽHTTP protocol compliance failed,Illegal meta character in value,Attack signature detected,Violation Rating Threat detected,Bot Client DetectedòUntrusted BotŠCriticalÂcurlÒ½
VIOL_ATTACK_SIGNATURE"™
	2000210944€Connection: keep-alive
Host: a.com
User-Agent: Java/1.7.0_51
Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2

"37*16"V
	2000110342?pt: */*
Foo: Authorization: %n%n%n%n
Foo: echo<!-- #echo
X18"29*6"W
	2000001793?7
Accept: */*
Foo: Authorization: %n%n%n%n
Foo: echo<!-- #ec"21*23"W
	2000041062?zation: %n%n%n%n
Foo: echo<!-- #echo
X1892: echo
X1893: '<!-"27*10